Partie 1

Exercice 1

In [1]:
import pandas as pd
import numpy as np

df = pd.read_csv('ventes_commerces.csv')
df.describe()

,produit_id,prix_unitaire,quantite,client_id,montant_total
count,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000
mean,10.136000,49.533070,2.973000,304.905000,146.310720
std,5.895953,19.522393,1.649543,114.567481,105.729758
min,1.000000,1.610000,0.000000,100.000000,0.000000
25%,5.000000,35.897500,2.000000,206.000000,69.065000
50%,10.000000,49.880000,3.000000,309.000000,127.910000
75%,15.000000,62.607500,4.000000,406.000000,199.792500
max,20.000000,104.620000,9.000000,499.000000,695.100000


In [2]:
df['montant_total'].sum()

146310.72

Exercice 2

In [3]:
vente_min = df['montant_total'].min()

transaction_min = df.loc[df["montant_total"].idxmin()]

print(transaction_min)


date             2023-02-24
produit_id               10
categorie            Meuble
prix_unitaire         39.86
quantite                  0
region               Centre
client_id               432
montant_total           0.0
Name: 54, dtype: object


In [4]:
vente_max = df['montant_total'].max()

transaction_max = df.loc[df["montant_total"].idxmax()]

print(transaction_max)

date             2023-03-06
produit_id                8
categorie          Vetement
prix_unitaire          99.3
quantite                  7
region               Centre
client_id               452
montant_total         695.1
Name: 64, dtype: object


Exercice 3

In [5]:
df['montant_total'].mean()

146.31071999999986

In [6]:
df['montant_total'].median()

127.91

Exercice 4

Une vente coute en moyenne 146,3 euros -> Moyenne
Il y a 50 % des ventes qui sont en dessous de 127.91 euros et 50% qui sont au dessus. -> Mediane

Partie 2

Exercice 1

In [7]:
df.groupby('categorie')['montant_total'].sum()

categorie
Alimentation    27977.18
Electronique    30207.55
Meuble          26966.78
Sport           30827.97
Vetement        30331.24
Name: montant_total, dtype: float64

Exercice 2

In [8]:
df.groupby('categorie')['montant_total'].sum().idxmax()

'Sport'

Exercice 3

In [9]:
df.groupby("categorie")["montant_total"].std()
      


categorie
Alimentation    106.453696
Electronique    109.615917
Meuble           91.609451
Sport           107.891508
Vetement        111.298651
Name: montant_total, dtype: float64

Partie 3

Exercice 1

In [10]:
print("\nMatrice de corrélation :")
colonnes_numeriques = ['prix_unitaire', 'quantite', 'montant_total']
matrice_corr = df[colonnes_numeriques].corr()
print(matrice_corr)


Matrice de corrélation :
               prix_unitaire  quantite  montant_total
prix_unitaire       1.000000 -0.029564       0.535673
quantite           -0.029564  1.000000       0.766617
montant_total       0.535673  0.766617       1.000000


Exercice 2

Plus la quantité est importante plus le montant total est élevé. 
Le montant total est influé directement par le prix unitaire

Exercice 3

En effet, certains outliers peuvent influencer cette corrélation